posthog/models/feedback/survey.py

```diff
@@ -0,0 +1,66 @@
+from django.db import models
+from posthog.models.utils import UUIDModel
+
+
+class Survey(UUIDModel):
+    class SurveyType(models.TextChoices):
+        POPOVER = "popover", "popover"
+        BUTTON = "button", "button"
+        EMAIL = "email", "email"
+        FULL_SCREEN = "full_screen", "full screen"
+
+    class Meta:
+        constraints = [models.UniqueConstraint(fields=["team", "name"], name="unique survey name for team")]
+
+    team: models.ForeignKey = models.ForeignKey(
+        "posthog.Team", on_delete=models.CASCADE, related_name="surveys", related_query_name="survey"
+    )
+    name: models.CharField = models.CharField(max_length=400)
+    description: models.TextField = models.TextField(blank=True)
+    linked_flag: models.ForeignKey = models.ForeignKey(
+        "posthog.FeatureFlag",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="surveys_linked_flag",
+        related_query_name="survey",
+    )
+    targeting_flag: models.ForeignKey = models.ForeignKey(
+        "posthog.FeatureFlag",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="surveys_targeting_flag",
+        related_query_name="survey",
+    )
+
+    responses_count: models.IntegerField = models.IntegerField(default=0)
```

Comment:
@neilkakkar Happy to take suggestions here on how to count or store a survey's responses count. It doesn't need to be stored on the model but it probably makes sense to keep a total count on this somewhere, should I just do this live with a fresh query each time the `surveys` page loads ? Or would that be too taxing?

Reply:
query each time survey page loads & cached on the API makes sense to me. Also, make it one request for all surveys, rather than one for each survey.

Probably doesn't make sense to have this on the model as it will change quickly and hard to keep in sync.